src/Symfony/Component/Messenger/Stamp/RetryCountStamp.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Stamp;
+
+/**
+ * Stamp applied by the transport to track retry count.
+ *
+ * @experimental in 4.3
+ */
+class RetryCountStamp implements StampInterface
+{
+    private $currentRetryCount;
+
+    public function __construct(int $currentRetryCount)
+    {
+        $this->currentRetryCount = $currentRetryCount;
+    }
+
+    public function getCurrentRetryCount(): int
```

Comment:
Minor comment: is it necessary to add the `Current` word here?  "RetryCount" cannot be mistaken with "MaxRetries"  ... and we don't use the generic (and confusing) "retries" anywhere.